protocol/shannon/sanctioned_endpoints_store.go

```diff
@@ -183,23 +183,137 @@ func (ses *sanctionedEndpointsStore) isSanctioned(
+
+// TODO_TECHDEBT(@commoddity,adshmh): update this to be composed of only
+//     - endpoint URL
+//     - either supplier address or session ID
+// Discord conversation: https://discord.com/channels/824324475256438814/1273320783547990160/1378346761151844465
+
+// sanctionKey:
 //   - Creates a unique key for session-based sanctions
 //   - Format: appAddr:sessionID:supplier:endpoint_url
-func sessionSanctionKey(
-	endpoint *endpoint,
-) string {
+type sanctionKey struct {
+	supplier    string
+	endpointURL string
+	appAddr     string
+	sessionID   string
+}
+
+func newSanctionKey(endpoint *endpoint) sanctionKey {
 	// Session header is never nil:
 	// Ref: https://github.com/pokt-network/shannon-sdk/blob/64d83f85e7e3f8e7d6ddee98ced276203cf5475f/session.go#L134
 	header := endpoint.session.Header
 
 	appAddr := header.ApplicationAddress
 	sessionID := header.SessionId
+	return sanctionKey{
+		supplier:    endpoint.supplier,
+		endpointURL: endpoint.url,
+		appAddr:     appAddr,
+		sessionID:   sessionID,
+	}
+}
+
+func sanctionKeyFromCacheKeyString(cacheKey string) sanctionKey {
+	// Only split for 4 parts, as final part is URL which contains a ":" character.
+	parts := strings.SplitN(cacheKey, ":", 4)
+	return sanctionKey{
+		appAddr:     parts[0],
+		sessionID:   parts[1],
+		supplier:    parts[2],
+		endpointURL: parts[3],
+	}
+}
+
+// string returns the string representation of the sanction key.
+// Example: "appAddr:sessionID:supplier:endpointURL"
+func (s sanctionKey) string() string {
 	return fmt.Sprintf(
 		"%s:%s:%s:%s",
-		appAddr,
-		sessionID,
-		endpoint.supplier,
-		endpoint.url,
+		s.appAddr,
+		s.sessionID,
+		s.supplier,
+		s.endpointURL,
+	)
+}
+
+func (s sanctionKey) endpointAddr() protocol.EndpointAddr {
+	return protocol.EndpointAddr(fmt.Sprintf("%s-%s", s.supplier, s.endpointURL))
+}
+
+// --------- Sanction Details ---------
+
+// getSanctionDetails returns the sanctioned endpoints for a given service ID.
+// It provides information about:
+//   - the currently sanctioned endpoints, including the reason
+//   - counts for valid and sanctioned endpoints
+//
+// It is called by the router to allow quick information about currently sanctioned endpoints.
+func (ses *sanctionedEndpointsStore) getSanctionDetails(serviceID protocol.ServiceID) devtools.ProtocolLevelDataResponse {
+	ses.logger.Info().Msgf("Getting sanction details for service ID: %s", serviceID)
+
+	permanentSanctionDetails := make(map[protocol.EndpointAddr]devtools.SanctionedEndpoint)
+	sessionSanctionDetails := make(map[protocol.EndpointAddr]devtools.SanctionedEndpoint)
+
+	// First get permanent sanctions
+	for key, sanction := range ses.permanentSanctions {
+		sanctionServiceID := protocol.ServiceID(sanction.sessionServiceID)
+
+		// Skip sanctions for service IDS other than the provided service ID
+		if sanctionServiceID != serviceID {
+			continue
+		}
+
+		ses.processSanctionIntoDetailsMap(string(key), sanction, permanentSanctionDetails)
```

Comment:
Should key be "endpointAddr"?

DItto below.

Reply:
no, it's a 4-component key. 

See TODO comment with linked Discord convo:
```go
// TODO_TECHDEBT(@commoddity,adshmh): update this to be composed of only
//     - endpoint URL
//     - either supplier address or session ID
// Discord conversation: https://discord.com/channels/824324475256438814/1273320783547990160/1378346761151844465
```